app/models/miq_action.rb

```diff
@@ -848,6 +848,42 @@ def action_container_image_analyze(action, rec, inputs)
     rec.scan
   end
 
+  def action_container_image_annotate_deny_execution(action, rec, inputs)
+    error_prefix = "MIQ(#{__method__}): Unable to perform action [#{action.description}], "
+    unless rec.kind_of?(ContainerImage)
+      MiqPolicy.logger.error("#{error_prefix} object [#{rec.inspect}] is not a Container Image")
+      return
+    end
+
+    unless rec.try(:ext_management_system).kind_of?(ManageIQ::Providers::Openshift::ContainerManagerMixin)
```

Comment:
Looks like something that should be refactored in the future.  We need to be able to define actions and action handler methods for each manager type and provider.

/cc @durandom